packages/plugin-ext/src/main/browser/webviews-main.ts

```diff
@@ -89,12 +90,19 @@ export class WebviewsMainImpl implements WebviewsMain, Disposable {
     }
 
     addOrReattachWidget(widget: WebviewWidget, showOptions: WebviewPanelShowOptions): void {
-        const widgetOptions: ApplicationShell.WidgetOptions = { area: showOptions.area ? showOptions.area : 'main' };
-
+        const area = showOptions.area ? showOptions.area : WebviewPanelTargetArea.Main;
+        const widgetOptions: ApplicationShell.WidgetOptions = { area };
         let mode = 'open-to-right';
-        if (showOptions.viewColumn === -2) {
-            const ref = this.shell.currentWidget;
-            if (ref && this.shell.getAreaFor(ref) === widgetOptions.area) {
+        // -2 corresponds to vscode.ViewColumn.Beside opening option
+        const canOpenBeside = showOptions.viewColumn === -2 && (area === WebviewPanelTargetArea.Main || area === WebviewPanelTargetArea.Bottom);
+        if (canOpenBeside) {
+            let activeOrRightmostTabbar = this.shell.getTabBarFor(area);
+            if (!activeOrRightmostTabbar) {
+                const areaTabBars = area === WebviewPanelTargetArea.Main ? this.shell.mainAreaTabBars : this.shell.bottomAreaTabBars;
+                activeOrRightmostTabbar = areaTabBars[areaTabBars.length - 1];
+            }
```

Comment:
Are there conditions under which this second attempt to get a tabbar works after the first has failed? My impression of the code behind `getTabBarFor` is that if there *is* a tabbar, it should be returned, but there are some subtleties to the tracking of `_currentTitle` in `theia-dock-panel.ts` that I don't follow completely.

Reply:
Yeah that's a good catch. I thought I had been getting `undefined` previously when a tabbar wasn't active, but that doesn't seem to be the case. I've removed the redundant check